src/mlpack/methods/ann/layer/recurrent_impl.hpp

```diff
@@ -125,6 +126,53 @@ Recurrent<InputDataType, OutputDataType, CustomLayers...>::Recurrent(
   this->network.push_back(recurrentModule);
 }
 
+template<typename InputDataType, typename OutputDataType,
+         typename... CustomLayers>
+size_t Recurrent<InputDataType, OutputDataType, CustomLayers...>::InputShape() const
+{
+  const size_t inputShapeStartModule = boost::apply_visitor(InShapeVisitor(), startModule);
+  // Return the input shape of the first module that we have.
+  if (inputShapeStartModule != 0)
+  {
+    return inputShapeStartModule;
+  }
+  // If input shape of first module is 0
```

Comment:
```suggestion
  // If input shape of first module is 0.
```

Missing stop at the end. There are some other comments below where this applies as well.